kotlin-eclipse-aspects/src/org/jetbrains/kotlin/aspects/ui/PackageExplorerLabelProviderAspect.aj

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.kotlin.aspects.ui;
+
+import org.eclipse.jdt.internal.ui.packageview.PackageExplorerContentProvider;
+import org.eclipse.jdt.internal.ui.packageview.PackageExplorerLabelProvider;
+import org.jetbrains.kotlin.ui.KotlinAwarePackageExplorerLabelProvider;
+
+@SuppressWarnings("restriction")
+public aspect PackageExplorerLabelProviderAspect {
+    
+    /**
+     * Replaces all instances of {@link PackageExplorerLabelProvider}
+     * with instances of {@link KotlinAwarePackageExplorerLabelProvider}
+     * 
+     * It affects classes {@link JavaNavigatorLabelProvider} and {@link PackageExplorerPart},
+     * which provides icons for Project Explorer and Package Explorer, respectively. 
+     */
+    PackageExplorerLabelProvider around(PackageExplorerContentProvider cp) 
+        : call(PackageExplorerLabelProvider.new(PackageExplorerContentProvider)) && args(cp) {
+        return new KotlinAwarePackageExplorerLabelProvider(cp);
+    }
```

Comment:
Alternatively, instead of overriding `PackageExplorerLabelProvider` constructor,
I could override `JavaElementImageProvider#getPackageFragmentIcon` which is used by `PackageExplorerLabelProvider`.

Then `KotlinAwarePackageExplorerLabelProvider` could be removed, and we would have something like that:
```java
ImageDescriptor around(IJavaElement element)
    : execution(ImageDescriptor JavaElementImageProvider.getPackageFragmentIcon(IJavaElement)) && args(element) {

    IPackageFragment fragment = (IPackageFragment)element;

    try {
        for (Object obj : fragment.getNonJavaResources()) {
            if (obj instanceof IFile) {
                String name = ((IFile) obj).getName();
                if (name.endsWith(".kt") || name.endsWith(".kts")) {
                    return JavaPluginImages.DESC_OBJS_PACKAGE;
                }
            }
        }
    }
    catch (JavaModelException e) { /* ignore */ }

    return proceed(element);
}
```